cmd/preview/list.go

```diff
@@ -73,15 +88,66 @@ func executeListPreviews(ctx context.Context, opts ListFlags) error {
 		return fmt.Errorf("failed to get preview environments: %s", err)
 	}
 	w := tabwriter.NewWriter(os.Stdout, 1, 1, 2, ' ', 0)
-	fmt.Fprintf(w, "Name\tScope\tSleeping\tLabels\n")
-	for _, preview := range previewList {
-		previewLabels := "-"
-		if len(preview.PreviewLabels) > 0 {
-			previewLabels = strings.Join(preview.PreviewLabels, ", ")
+	switch opts.output {
+	case "json":
+		var previewSlice []PreviewJson
+		for _, preview := range previewList {
+			previewLabels := "-"
+			if len(preview.PreviewLabels) > 0 {
+				previewLabels = strings.Join(preview.PreviewLabels, ", ")
+			}
+			previewJson := PreviewJson{
+				Name:     preview.ID,
+				Scope:    preview.Scope,
+				Sleeping: preview.Sleeping,
+				Labels:   previewLabels,
+			}
+			previewSlice = append(previewSlice, previewJson)
+		}
+		bytes, err := json.MarshalIndent(previewSlice, "", " ")
+		if err != nil {
+			return err
+		}
+		oktetoLog.Println(string(bytes))
+	case "yaml":
+		var previewSlice []PreviewJson
+		for _, preview := range previewList {
+			previewLabels := "-"
+			if len(preview.PreviewLabels) > 0 {
+				previewLabels = strings.Join(preview.PreviewLabels, ", ")
+			}
+			previewJson := PreviewJson{
+				Name:     preview.ID,
+				Scope:    preview.Scope,
+				Sleeping: preview.Sleeping,
+				Labels:   previewLabels,
+			}
+			previewSlice = append(previewSlice, previewJson)
+		}
+		bytes, err := yaml.Marshal(previewSlice)
+		if err != nil {
+			return err
+		}
+		oktetoLog.Println(string(bytes))
+	default:
+		fmt.Fprintf(w, "Name\tScope\tSleeping\tLabels\n")
+		for _, preview := range previewList {
+			previewLabels := "-"
+			if len(preview.PreviewLabels) > 0 {
+				previewLabels = strings.Join(preview.PreviewLabels, ", ")
+			}
+			fmt.Fprintf(w, "%s\t%s\t%v\t%s\n", preview.ID, preview.Scope, preview.Sleeping, previewLabels)
 		}
-		fmt.Fprintf(w, "%s\t%s\t%v\t%s\n", preview.ID, preview.Scope, preview.Sleeping, previewLabels)
 	}
-
 	w.Flush()
```

Comment:
if we output in JSON/YAML, is this `w.Flush()` really needed?

Reply:
without `w.Flush()` the default behaviour will not be executed. However, we don't need it for `json` or `yaml`

Reply:
@nitishfy I suggest to move all tablewriter logic inside the default case

Reply:
Resolved